src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -219,9 +224,10 @@ class ImplLookup(
         }
     }
 
-    private fun assembleCandidates(ref: TraitRef): List<SelectionCandidate> = buildList {
+    private fun assembleCandidates(ref: TraitRef): Set<SelectionCandidate> = buildSet {
```

Comment:
Same question about `Set`, but here, I guess, the point is that we can somehow run into multiple implementations of `Sized` trait (but I don't understand how). Anyway, think it's ok for now because I going to optimize this method (i.e. why we need to call the costly `assembleImplCandidates` for type parameters?), and we can handle `Sized` trait as a special case in the same manner (i.e. why we need to call the costly `assembleImplCandidates` if we selecting `Sized` trait that can't be implemented manually?).

Reply:
Very stupid case is
```rust
fn foo<T: Trait>(t: T) where T: Trait { ... }
```
It looks like valid rust function for any `Trait`.

My case is
```rust
fn foo<T: Sized>(t: T) { ... }
```
In this case `Sized` trait comes from explicit type bound and from `assembleImplicitCandidates`. Of course, I can check this case manually but I think `Set`
 can solve both problems more elegantly.

Reply:
Hm, if we want to have special case for `Sized` trait in `assembleCandidates` I can implement it in this PR - we already have `Ty.isSized` function and we use it in `assembleCandidates` directly for `Sized` trait case. What do you think about it?

Reply:
Ok